### PR TITLE
Issue in build scan result_url when JFrog URL is not configured

### PR DESCRIPTION
**Pull request by dortam888**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----
Fixed an issue: when JFrog URL is not configured but ArtifactoryURL and XrayURL, the scan_url is returned wrongly.
See discussion in https://github.com/jfrog/jfrog-cli/issues/2629

### CODE REVIEW

File: commands/scan/buildscan.go [hunk 1]
```diff
@@ -186,7 +191,21 @@ func (bsc *BuildScanCommand) CommandName() string {
 	return "xr_build_scan"
 }
 
-func trimBuildScanResultUrl(fullUrl string) (string, string, error) {
+// There are two cases. when serverDetails.Url is configured and when serverDetails.XrayUrl and serverDetails.ArtifactoryUrl are configured
+// The function will retutn the Url if configured and will trim xray if serverDetails.Url is not configured
+func (bsc *BuildScanCommand) configuredCliUrl() (string, error) {
```
Comment (attiasas) [inline commands/scan/buildscan.go:196]: you can create a different method. `func getActualUrl(config.ServerDetails) (string, error)` and than add test for this

**Review by attiasas — CHANGES_REQUESTED** [review]

File: commands/scan/buildscan.go [hunk 2]
```diff
@@ -186,7 +191,21 @@ func (bsc *BuildScanCommand) CommandName() string {
 	return "xr_build_scan"
 }
 
-func trimBuildScanResultUrl(fullUrl string) (string, string, error) {
+// There are two cases. when serverDetails.Url is configured and when serverDetails.XrayUrl and serverDetails.ArtifactoryUrl are configured
+// The function will return the Url if configured and will trim xray if serverDetails.Url is not configured
+func getActualUrl(serverDetails config.ServerDetails) (string, error) {
+	url := serverDetails.Url
```
Comment (attiasas) [inline commands/scan/buildscan.go:197]:
```suggestion
	url := trimUrl(serverDetails.Url)
```
should we trim `serverDetails.Url` as well?

File: commands/scan/buildscan_test.go [hunk 3]
```diff
@@ -3,26 +3,49 @@ package scan
...
+	expectedUrl := "http://dort.jfrog.io/"
+	actualUrl, err := getActualUrl(*buildScanCommand.serverDetails)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedUrl, actualUrl)
+
+	// test when only Artifactory URL and Xray URL are provided and URL is empty
+	buildScanCommand.serverDetails.Url = ""
+	actualUrl, err = getActualUrl(*buildScanCommand.serverDetails)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedUrl, actualUrl)
```
Comment (attiasas) [inline commands/scan/buildscan_test.go:50]:
```suggestion
        expectedUrl := "http://dort.jfrog.io/"
        testCases := []struct {
		name      string
		serverDetails config.ServerDetails
        }{
               {
                     name: "No JFrog URL",
                     serverDetails: config.ServerDetails{
		          Url:            "http://dort.jfrog.io/",
		          ArtifactoryUrl: "http://dort.jfrog.io/artifactory",
		          XrayUrl:        "http://dort.jfrog.io/xray",
	             },
               },
               {
                     name: "JFrog URL is provided",
                     serverDetails: config.ServerDetails{
		          ArtifactoryUrl: "http://dort.jfrog.io/artifactory",
		          XrayUrl:        "http://dort.jfrog.io/xray",
	             },
               },
        }
        
        for _, tc := range testCases {
              t.Run(tc.name, func(t *testing.T) {
                     actualUrl, err = getActualUrl(tc.serverDetails)
	             assert.NoError(t, err)
	             assert.Equal(t, expectedUrl, actualUrl)
              })
        }
```
When you have more than one test case, you should implement with minimum code duplications

**Review by attiasas — CHANGES_REQUESTED** [review]

**Review by attiasas — APPROVED** [review]
